src/dh/dhe.ts

```diff
@@ -0,0 +1,235 @@
+  const url = new URL('/socket', serverUrl);
+  if (url.protocol === 'http:') {
+    url.protocol = 'ws:';
+  } else {
+    url.protocol = 'wss:';
+  }
+  return url;
+}
+
+/**
+ * Determine if the logged in user has permission to interact with the UI.
+ * @param dheClient The DHE client.
+ * @returns A promise that resolves to true if the user has permission to interact with the UI.
+ */
+export async function hasInteractivePermission(
+  dheClient: EnterpriseClient
+): Promise<boolean> {
+  // TODO: Retrieve these group names from the server:
+  // https://deephaven.atlassian.net/browse/DH-9418
+  const GROUP_NON_INTERACTIVE = 'deephaven-noninteractive';
+  const GROUP_SUPERUSERS = 'iris-superusers';
+
+  const groups = await dheClient.getGroupsForUser();
+
+  const isSuperUser = groups.indexOf(GROUP_SUPERUSERS) >= 0;
+  const isNonInteractive = groups.indexOf(GROUP_NON_INTERACTIVE) >= 0;
+  const isInteractive = !isNonInteractive;
+
+  return isSuperUser || isInteractive;
+}
+
+/**
+ * Find the `QueryInfo` matching the given serial.
+ * @param dheClient DHE client to search.
+ * @param matchSerial Serial to match.
+ * @returns The matching `QueryInfo` or `undefined` if not found.
+ */
+export function findQueryConfigForSerial(
+  dheClient: EnterpriseClient,
+  matchSerial: string
+): QueryInfo | undefined {
+  return dheClient
+    .getKnownConfigs()
+    .find(({ serial }) => serial === matchSerial);
+}
+
+/**
+ * Create a query of type `InteractiveConsole`.
+ * @param tagId Unique tag id to include in the query name.
+ * @param dheClient The DHE client to use to create the query.
+ * @param consoleType The type of console to create.
+ * @returns A promise that resolves to the serial of the created query. Note
+ * that this will resolve before the query is actually ready to use. Use
+ * `getWorkerInfoFromQuery` to get the worker info when the query is ready.
+ */
+export async function createInteractiveConsoleQuery(
+  tagId: UniqueID,
+  dheClient: EnterpriseClient,
+  consoleType?: ConsoleType
+): Promise<QuerySerial> {
+  const userInfo = await dheClient.getUserInfo();
+  const owner = userInfo.username;
+  const type = INTERACTIVE_CONSOLE_QUERY_TYPE;
+
+  const [dbServers, queryConstants, serverConfigValues] = await Promise.all([
+    dheClient.getDbServers(),
+    dheClient.getQueryConstants(),
+    dheClient.getServerConfigValues(),
+  ]);
+
+  const name = `IC - VS Code - ${tagId}`;
+  const dbServerName = dbServers[0]?.name ?? 'Query 1';
+  const heapSize = queryConstants.pqDefaultHeap;
+  const jvmProfile = serverConfigValues.jvmProfileDefault;
+  const scriptLanguage =
+    serverConfigValues.scriptSessionProviders?.find(
+      p => p.toLocaleLowerCase() === consoleType
+    ) ?? 'Python';
+  const workerKind = serverConfigValues.workerKinds?.[0]?.name;
+
+  const timeZone =
+    serverConfigValues.timeZone ??
+    Intl.DateTimeFormat().resolvedOptions().timeZone;
+
+  const scheduling = QueryScheduler.makeDefaultScheduling(
```

Comment:
@vbabich 
Looks like that calls:https://docs.deephaven.io/pycoreplus/20240517/_modules/deephaven_enterprise/client/generate_scheduling.html#GenerateScheduling.generate_temporary_scheduler

Does that mean instead of calling `makeDefaultScheduling`, we'd want to add a new method like `makeTemporaryScheduling` and return an array of key=value tokens similar to `generate_temporary_scheduler`?

Reply:
I've created a PR based on the referenced code that "should" give us a comparable temporary schedule in the JS client:
https://github.com/deephaven-ent/iris/pull/2348

Reply:
Yes, `makeDefaultScheduling` generates the scheduling config for new Script type queries  I'd add a function for temporary scheduling.